src/actions/messages/MESSAGE_UPDATE.ts

```diff
@@ -0,0 +1,26 @@
+import { Action } from '../../lib/structures/Action';
+import { isTextBasedChannel } from '../../util/Util';
+import { extender } from '../../util/Extender';
+
+import type { MessageUpdateDispatch } from '@klasa/ws';
+import type { Message } from '../../client/caching/structures/Message';
+
+export default class CoreAction extends Action {
+
+	public check(data: MessageUpdateDispatch): Message | null {
+		const guild = data.d.guild_id ? this.client.guilds.get(data.d.guild_id) : undefined;
+		const channel = guild ? guild.channels.get(data.d.channel_id) : this.client.channels.get(data.d.channel_id);
+		if (typeof channel === 'undefined') return null;
+		if (!isTextBasedChannel(channel)) return null;
+		return channel.messages.get(data.d.id) ?? null;
+	}
+
+	public build(data: MessageUpdateDispatch): Message | null {
+		return new (extender.get('Message'))(this.client, data.d);
```

Comment:
What should I do instead?

Reply:
This is the part where we decide how we handle partial data, and same should be done here. CC @bdistin

Reply:
We talked about emitting XPartial for partials, as a separate instance, not cached. So like MessageCreatePartial would emit a new PartialMessage that's not in the cache, and all it is is enough data to fetch the instance (with a boolean cache option). To be clear, Message would not extend PartialMessage as that would add a lot of nullish checks that aren't fun or useful.

Reply:
Not really, because types would be differently.

One way to ensure data immutability (thus increasing predictability) is making `PartialMessage#fetch` return an instance of `Message` instead of mutating itself, that way we promote objects without punishing TypeScript users.

Reply:
Yes to the first part, no to the second. We shouldn't be sending Message | PartialMessage over the same event, because that just re-introduces checks to see what's what. When we do PartialMessage#fetch, it will likely emit over MessageCreate again, since we are newly caching a full message (if enabled).

Reply:
Emitting `messageCreate` from a messageUpdate makes no sense, the message wasn't just created, it was fetched/recreated..soo

Reply:
See the discussion channel for further conversations.

Reply:
Resolved in bdb037d 🎉